lib/scenic/adapters/postgres.rb

```diff
@@ -226,6 +264,35 @@ def raise_unless_concurrent_refresh_supported
           raise ConcurrentRefreshesNotSupportedError
         end
       end
+
+      def indexes_for(tbl)
+        execute "SELECT indexname FROM pg_indexes WHERE tablename = '#{tbl}'"
+      end
+
+      def update_index_names_for(old_tbl, tbl)
+        indexes_for(tbl).map { |r| r["indexname"] }.each do |indexname|
+          new_indexname = indexname.gsub(old_tbl, tbl)
+
+          execute "ALTER INDEX #{indexname} RENAME TO #{new_indexname}" unless indexname == new_indexname
```

Comment:
Please don't use 'conditional surprises', preferring the block format of conditions. Also please invert the conditional to `if index_name != new_index_name`